### ISSUE TITLE
Why are there different results in tensorflow and pytorch?

### ISSUE THREAD

**Issue by shanqq377** [anibali/dsnt-pose2d#3]
in tensorflow:
`def _test_reg_loss(shift_mean=True):
    # Target mean and standard deviation
    #target_mean = torch.Tensor([0, 0])
    target_mean = tf.constant([[0., 0.]])
    target_stddev = 0.4

    # Helper function to calculate the loss between the target and a Gaussian heatmap
    # parameterized by `mean` and `stddev`.
    def calc_loss(mean, stddev):
        hm = dsnt._make_gaussians(mean, 5, 5, fwhm=stddev)
        return dsnt.js_reg_loss(hm, target_mean, target_stddev)

    # Minimum loss occurs when the heatmap's mean and standard deviation are the same
    # as the target
    min_loss = calc_loss(target_mean, target_stddev)
    with tf.Session() as sess:
        print(min_loss.eval())
        print(calc_loss(target_mean, target_stddev + 0.2).eval())
        print(calc_loss(target_mean, target_stddev - 0.2).eval())

_test_reg_loss()`
![2](https://user-images.githubusercontent.com/49785064/75552386-295d3500-5a71-11ea-8815-4940e294aaa9.PNG)

in pytorch:
`def _test_reg_loss(tc, loss_method, shift_mean=True):
    # Target mean and standard deviation
    target_mean = torch.Tensor([[0., 0.]])
    target_stddev = 0.4

    # Helper function to calculate the loss between the target and a Gaussian heatmap
    # parameterized by `mean` and `stddev`.
    def calc_loss(mean, stddev):
        hm = make_gauss(mean, 5, 5, sigma=stddev)
        return loss_method(hm, target_mean, target_stddev, mask=None)

    # Minimum loss occurs when the heatmap's mean and standard deviation are the same
    # as the target
    min_loss = calc_loss(target_mean, target_stddev)
    print(min_loss)
    # Minimum loss should be close to zero
    tc.assertEqual(min_loss, torch.tensor(0), 1e-3)

    # Loss should increase if the heatmap has a larger or smaller standard deviation than
    # the target
    print(calc_loss(target_mean, target_stddev + 0.2))
    print(calc_loss(target_mean, target_stddev - 0.2))
    tc.assertGreater(calc_loss(target_mean, target_stddev + 0.2), min_loss + 1e-3)
    tc.assertGreater(calc_loss(target_mean, target_stddev - 0.2), min_loss + 1e-3)`

![55](https://user-images.githubusercontent.com/49785064/75552349-12b6de00-5a71-11ea-87f8-46a7c6845b3e.PNG)

**Comment by anibali** [comment]
I didn't write the Tensorflow implementation, but from what I know it represents coordinates in the (0, 1) range, as opposed to the (-1, 1) range used here. So if the standard deviation is in different units, then the results will be different (ie +-0.2 will have a different effect). If you want to dig deeper, save and compare the target gaussians from each implementation.

**Comment by shanqq377** [comment]
Thank you. I've found that gaussians transformation is different.
